core/src/signer.ts

```diff
@@ -0,0 +1,30 @@
+import { ethers } from 'ethers';
+import NETWORKS from './constants/NETWORKS';
+import getProvider from './provider';
+
+const signers: Record<string, ethers.Wallet> = {};
+
+const getSigner = function (network: string): ethers.Wallet | undefined {
+    if (!NETWORKS[network]) {
+        throw new Error(`The network "${network}" is not supported yet!`);
+    }
+
+    if (!signers[network]) {
+        return undefined;
+    }
+    return signers[network];
+};
+
+export const setSigner = async function (network: string, privateKey: string) {
+    const provider = getProvider(network);
+
+    try {
+        signers[network] = new ethers.Wallet(privateKey, provider);
+        const address = await signers[network].getAddress();
+        console.info(`Authenticated with wallet "${address}"`);
+    } catch (error) {
+        console.error(`Could not authenticate with the private key: ${error}`);
```

Comment:
Should we instead re-throw the error here? How else would the parent function calling `setSigner` know that it failed?